lms/djangoapps/edraak_university/tests/tests.py

```diff
@@ -0,0 +1,480 @@
+        }
+
+        self.assertIn('university_id', course_tabs,
+                      'Course tab is not installed, run `$ pip install -r requirements/edx/local.txt` '
+                      'to solve the problem')
+
+        tab_class_name = course_tabs['university_id'].type
+        self.assertEquals(tab_class_name, 'university_id', 'Should have the required tab, with a correct tab.type')
+
+
+@ddt.ddt
+class UniversityTabIDTest(ModuleStoreTestCase):
+    """
+    Unit and integration tests for the tab.
+    """
+
+    # This patch disables `SafeCookieData.parse` able to use client.login() without strict checking and errors
+    # Other than this, it is not needed in our tests
+    CUSTOM_MIDDLEWARE_CLASSES = [
+        'django.contrib.sessions.middleware.SessionMiddleware' if cls.endswith('SafeSessionMiddleware') else cls
+        for cls in settings.MIDDLEWARE_CLASSES
+    ]
+
+    def test_xmodule_field(self):
+        course = CourseFactory.create()
+
+        self.assertTrue(hasattr(course, 'enable_university_id'),
+                        'The course should have an xmodule enable_university_id field in its field definitions')
+
+        self.assertFalse(course.enable_university_id, 'The feature should be disabled on courses by default')
+
+    @ddt.data(True, False)
+    def test_is_enabled_not_logged_in(self, enable_university_id):
+        course = CourseFactory.create()
+        course.enable_university_id = enable_university_id
+
+        tab = UniversityIDTab(tab_dict={})
+        self.assertFalse(tab.is_enabled(course, user=None), 'Should be disabled for all non-logged-in users')
+
+    @ddt.data(True, False)
+    def test_is_enabled_not_enrolled(self, enable_university_id):
+        user = UserFactory.create()
+        course = CourseFactory.create()
+        course.enable_university_id = enable_university_id
+
+        tab = UniversityIDTab(tab_dict={})
+        self.assertFalse(tab.is_enabled(course, user), 'Should be disabled for all non-enrolled')
+
+    @ddt.unpack
+    @ddt.data(
+        {'course_enable_university_id': False, 'should_tab_be_enabled': False},
+        {'course_enable_university_id': True, 'should_tab_be_enabled': True},
+    )
+    def test_is_enabled_enrolled(self, course_enable_university_id, should_tab_be_enabled):
+        user = UserFactory.create()
+        course = CourseFactory.create()
+        course.enable_university_id = course_enable_university_id
+
+        with patch.object(CourseEnrollment, 'is_enrolled', return_value=True):
+            tab = UniversityIDTab(tab_dict={})
+            self.assertEqual(tab.is_enabled(course, user), should_tab_be_enabled,
+                             'Should only be enabled when `enable_university_id` is, even for enrolled users')
+
+            with patch.dict(settings.FEATURES, EDRAAK_UNIVERSITY_APP=False):
+                self.assertEqual(
+                    first=tab.is_enabled(course, user),
+                    second=False,
```

Comment:
by convention, ```first=``` and ```second=``` are not common in edX tests, so no need to use I think, not a big deal though, if you want to keep them then do.